### ISSUE TITLE
ALLOW_EMPTY_STRING_KEYS doesn't work when skip=True for dpath.util.get in version 1.5.0 (reopening of issue #82)

### ISSUE THREAD

**Issue by srids** [dpath-maintainers/dpath-python#118]
Though I see issue #82 is closed and merged for version 1.5.0 (via issue #73 via pull request #96) - am still seeing this issue in version 1.5.0

```
import dpath

dpath.version.VERSION
'1.5.0'

dpath.options.ALLOW_EMPTY_STRING_KEYS=True
tdict = {'': 'fefa1c8e-9c9f-4da4-8964-a7d93274c101'}
dpath.util.get(tdict, '')

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/site-packages/dpath/util.py", line 120, in get
    matches = list(search(obj, glob, yielded=True, separator=separator))
  File "/usr/local/lib/python3.6/site-packages/dpath/util.py", line 158, in _search_yielded
    globlist = __safe_path__(glob, separator)
  File "/usr/local/lib/python3.6/site-packages/dpath/util.py", line 33, in __safe_path__
    key = elem[0]
IndexError: string index out of range
```

or with a similar test as

```
import dpath

dpath.version.VERSION
'1.5.0'

dpath.options.ALLOW_EMPTY_STRING_KEYS=True
tdict = {
    "Empty": {
        "": {
            "Key": ""
        }
    }
}
dpath.util.get(tdict, 'Empty//Key')

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/site-packages/dpath/util.py", line 120, in get
    matches = list(search(obj, glob, yielded=True, separator=separator))
  File "/usr/local/lib/python3.6/site-packages/dpath/util.py", line 158, in _search_yielded
    globlist = __safe_path__(glob, separator)
  File "/usr/local/lib/python3.6/site-packages/dpath/util.py", line 33, in __safe_path__
    key = elem[0]
IndexError: string index out of range
```

**Comment by calebcase** [comment]
Confirmed this is not an issue in v2.0.1. We are investigating older 1.x version now to see if it is a recent introduction.

**Comment by akesterson** [comment]
Turns out this bug has been present in 1.x since at least 1.3.2. @srids Can you check #124 to see if it resolves your issue? The fix works for me.

**Comment by akesterson** [comment]
Blocked by #136

**Comment by srids** [comment]
> 

@akesterson so change in #124 works. I see this has been merged into 1.x branch. Any way to cut a 1.x version and push it to pypi? Thanks.
